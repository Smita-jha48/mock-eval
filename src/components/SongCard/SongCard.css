.card-body{
    width: 250px;
    height: 250px;
    gap: 10px;
    border-radius: 15px;
    box-shadow: 2px 2px 5px gray;
}
.card-img{
    border-radius: 50%;
    width: 170px;
    margin: 20px 20px 0px 40px;
}
.song-name{
   height: 25px;
   width: 150px;
   white-space: nowrap; 
   overflow: hidden;
   text-overflow: ellipsis; 
    
}
.left-section {
    flex-direction: column;
}
.card-footer{
    display: -webkit-flex;
    justify-content: space-around;
    margin-bottom: 10px;
}

.heart-img{
    margin-bottom: 5px;
}

.heart-count{
 position: relative;
  text-align: center;
  color: white;
}
.count{
  position: absolute;
  top: 40%;
  left: 50%;
  transform: translate(-50%, -50%);
}