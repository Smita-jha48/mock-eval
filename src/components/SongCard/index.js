import React, {useState, useEffect} from 'react';
import heartGrey from '../../assets/Image/heart-gray.svg';
import heartRed from '../../assets/Image/heart-red.svg';
import './SongCard.css'

function SongCard({id, index, countLike, data, handleLike, bgColor}){

    return (
       <div className='card-body ' style={{ backgroundColor: bgColor }}>
       <img className='card-img' src={data.imageUrl} alt='song'></img>
       <div className='card-footer '>
       <div className='left-section '>
       <p className='song-name'>{data.name}</p>
       <h4>{data.artist.name}</h4>
       </div>
       <div className='heart-count'>
       <img src={countLike.like? heartRed:heartGrey} alt='heart' className='heart-img' onClick={()=>handleLike(id, index)}></img>
       <h6 className='count'>{countLike.count}</h6>
       </div>
       </div>
       </div>
    )
}

export default SongCard;