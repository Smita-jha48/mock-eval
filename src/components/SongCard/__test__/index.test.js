import { fireEvent, render, screen, waitFor } from '@testing-library/react';
import heartGrey from '../../../assets/Image/heart-gray.svg'

import SongCard from '..';

describe('song card', ()=>{
    const mockProps = {
        id:1, 
        index:1, 
        countLike:{
        "count": 1,
        "like": false
        }, 
        data:{
                "id": "cd3cc1e3-e1e0-4ccd-bc67-747648985838",
                "name": "Lost",
                "imageUrl": "https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631",
                "artist": {
                    "id": "496b0a85-2bfa-45bc-8d0f-57fe0ce55708",
                    "name": "Maroon 5"
                },
                "genre": {
                    "id": "128aa7f8-c943-48ce-b352-7edd26fa4c6e",
                    "name": "Pop"
                }
            }, 
        handleLike:jest.fn(), 
      };
      it('should render correctly', () => {
        const { asFragment } = render(<SongCard {...mockProps} />);
        expect(asFragment()).toMatchSnapshot();
      });
      it('should render with heart image', () => {
        const { getByAltText } = render(<SongCard {...mockProps} />);
        const imgElement = getByAltText('heart');
        expect(imgElement.getAttribute('src')).toBe(heartGrey);
      });
      it('should render with heart count', () => {
        const { getByText } = render(<SongCard {...mockProps} />);
        const heartCountElement = getByText(mockProps.countLike.count);
        expect(heartCountElement).toBeTruthy();
      });
      it('handleLike should be called on click', async () => {
        const { getByAltText } = render(<SongCard {...mockProps} />);
        const imgElement = getByAltText('heart');
        fireEvent.click(imgElement);
        waitFor(() => {
          expect(mockProps.handleLike).toHaveBeenCalled();
        });
      });
})