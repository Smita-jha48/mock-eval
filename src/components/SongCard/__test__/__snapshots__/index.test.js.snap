// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`song card should render correctly 1`] = `
<DocumentFragment>
  <div
    class="card-body "
  >
    <img
      alt="song"
      class="card-img"
      src="https://i.scdn.co/image/ab67616d0000b27386a8ab515de4b7aef28cd631"
    />
    <div
      class="card-footer "
    >
      <div
        class="left-section "
      >
        <p
          class="song-name"
        >
          Lost
        </p>
        <h4>
          Maroon 5
        </h4>
      </div>
      <div
        class="heart-count"
      >
        <img
          alt="heart"
          class="heart-img"
          src="heart-gray.svg"
        />
        <h6
          class="count"
        >
          1
        </h6>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
